libs/polysemy-wire-zoo/src/Wire/Sem/Now/MonadIO.hs

```diff
@@ -0,0 +1,53 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
+
+module Wire.Sem.Now.MonadIO
+  ( nowToMonadIOAction,
+    nowToMonadIO,
+  )
+where
+
+import Data.Time
+import Imports
+import Polysemy
+import Wire.Sem.FromUTC
+import Wire.Sem.Now
+
+-- | An interpreter of the 'Now' effect to MonadIO via a custom IO action that
+-- provides the current time.
+nowToMonadIOAction ::
+  forall m r a.
+  (MonadIO m, Member (Embed m) r) =>
```

Comment:
What's the value of doing this in `MonadIO` rather than in `IO` directly? `MonadIO` doesn't make much sense in the land of effects.

Reply:
Oh, OK, I didn't know that. I was just trying to make it as polymorphic as possible. I'll revert this back to IO.